onejs/index.ts

```diff
@@ -76,4 +72,57 @@ export function useEvent<TEventName extends string, TCallback>(
             unregisterOnEngineReload(removeHandler)
         }
     }, dependencies)
-}
\ No newline at end of file
+}
+
+/**
+ * Describes a C# class or struct that contains a property that is a C# event.
+ * For example, given a C# class that declares an event called `OnValueChanged`
+ * whose delegates accept a single parameter of type `int`, you can declare the
+ * type as follows:
+ *
+ *   type MyType = HasCsharpEvent<"OnValueChanged", number> & {
+ *     // other properties...
+ *   }
+ * 
+ * For event delegates that take more than one value, see related types
+ * `HasCsharpEvent2` and `HasCsharpEvent3`.
+ */
+export type HasCsharpEvent<EventName extends string, TVal> = HasCSharpEventBase<
+  EventName,
+  (val: TVal) => void
+>;
+
+export type HasCsharpEvent2<
+  EventName extends string,
+  TVal1,
+  TVal2
+> = HasCSharpEventBase<EventName, (val1: TVal1, val2: TVal2) => void>;
+
+export type HasCsharpEvent3<
+  EventName extends string,
+  TVal1,
+  TVal2,
+  TVal3
+> = HasCSharpEventBase<
+  EventName,
+  (val1: TVal1, val2: TVal2, val3: TVal3) => void
+>;
+
+/**
+ * A type that describes a C# class or struct that contains the following
+ * properties/fields that conform to the useEventfulState() protocol:
+ * 
+ * - a property with an arbitrary name and type
+ * - a C# event named `On{PropertyName}Changed`, whose delegate accepts a single
+ *   parameter of the same type as the property
+ */
+export type HasEventfulProperty<PropName extends string, TVal> = Record<
```

Comment:
I wonder if `HasEventfulState` is a better name here, given that it's meant to be used with `useEventfulState`.

Reply:
The emphasis on Property from `HasEventfulProperty` may also be good for specificity

Reply:
I don't have a strong opinion, so I'll keep it as-is.